hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/streams/StreamValidationOp.java

```diff
@@ -55,44 +62,96 @@ public class StreamValidationOp extends UtilOp {
                                     .flatMap(v -> v.validationErrorsIn(data))
+                                    .peek(t -> log.error("Record stream validation error!", t))
                                     .map(Throwable::getMessage)
                                     .collect(joining(ERROR_PREFIX));
                             if (!maybeErrors.isBlank()) {
                                 throw new AssertionError(
                                         "Record stream validation failed:" + ERROR_PREFIX + maybeErrors);
                             }
+                            dataRef.set(data);
                         },
                         () -> Assertions.fail("No record stream data found"));
+        // Freeze the network
+        allRunFor(
+                spec,
+                freezeOnly().payingWith(GENESIS).startingIn(2).seconds(),
+                // Wait for the final stream files to be created
+                sleepFor(8 * BUFFER_MS));
+        readMaybeBlockStreamsFor(spec)
+                .ifPresentOrElse(
+                        blocks -> {
+                            final var data = requireNonNull(dataRef.get());
+                            // TODO - append the final record file to the record stream data
+                            final var maybeErrors = BLOCK_STREAM_VALIDATOR_FACTORIES.stream()
+                                    .filter(factory -> factory.appliesTo(spec))
+                                    .map(factory -> factory.create(spec))
+                                    .flatMap(v -> v.validationErrorsIn(blocks, data))
+                                    .map(Throwable::getMessage)
+                                    .collect(joining(ERROR_PREFIX));
+                            if (!maybeErrors.isBlank()) {
+                                throw new AssertionError(
+                                        "Block stream validation failed:" + ERROR_PREFIX + maybeErrors);
+                            }
+                        },
+                        () -> {});
         return false;
     }
 
-    private static Optional<RecordStreamAccess.Data> readMaybeStreamDataFor(@NonNull final HapiSpec spec) {
+    private static Optional<List<Block>> readMaybeBlockStreamsFor(@NonNull final HapiSpec spec) {
+        List<Block> blocks = null;
+        final var blockPaths = spec.getNetworkNodes().stream()
+                .map(node -> node.getExternalPath(BLOCK_STREAMS_DIR))
+                .map(Path::toAbsolutePath)
+                .toList();
+        for (final var path : blockPaths) {
+            try {
+                log.info("Trying to read blocks from {}", path);
+                blocks = BLOCK_STREAM_ACCESS.readBlocks(path);
+                log.info("Read {} blocks from {}", blocks.size(), path);
+            } catch (Exception ignore) {
+                // We will try to read the next node's streams
+            }
```

Comment:
We should at least log the exception here. When I ran this test, if there was ever an incomplete record file, a catch block like this swallowed that exception and reported that there were no record files, even though record files existed

Reply:
I think its intentionally ignored. @tinker-michaelj ?